README.md

```diff
@@ -1,634 +1,1828 @@
+    - The loan must still contain the Borrower ID.
+    - Both the Lock and Key NFT for this loan must be burned.
+    - All active beacons among the inputs must be burned.
+
+The invalid-before bound is used to prove the expiration time has passed. When claiming multiple
+expired loans, invalid-before can be set to the latest loan expiration.
+
+The Borrower ID is required so that only the borrower can claim any remaining collateral in the UTxO
+as well as the minUTxOValue deposit that was stored with the Active UTxO. This deposit belongs to
+the borrower. Finished loans should not be claimable with the Key NFT for this reason.
+
+Since the Key NFT is actually being burned in this transaction (ie, the UTxO is being consumed),
+this can be used as a proxy for the lender's approval.
+
+All loans claimed will count as defaults against the borrower.
+
+If the claim period has passed, the owner of the Key NFT can still claim the expired collateral.
+However, the borrower is also now able to claim the collateral so the protocol will give the
+collateral to whoever is first. Lenders should claim any collateral before the `claimExpiration`
+passes. The claim period was part of the negotiations so this shouldn't be a problem.
+
+##### Unlocking Finished/Lost Loans
+
+> NOTE: In the future, the step of unlocking finished loans may not be needed.
+
+A borrower is able to clean up any Finished Active UTxOs as well as claim any expired Active UTxOs
+whose claim period has passed.
+
+To unlock Active UTxOs, all of the following must be true:
+
+- The Active UTxO being unlocked must have an `ActiveDatum`.
+- The Active UTxO being unlocked must be spent using the `Unlock` redeemer.
+- All Active UTxOs being unlocked must come from the same loan address.
+- If any of the Active UTxOs spent with `Unlock` contain beacons, the active smart contract must be
+executed as a minting policy using the `BurnRemainderOrUnlockLost` redeemer.
+- For all Active UTxOs being unlocked:
+    - If the Borrower ID is still present, the loan's `claimExpiration` must have passed (using
+    invalid-before).
+    - All active beacons among the inputs must be burned.
+- The loan address' staking credential must signal approval.
+
+The invalid-before bound is used to prove the claim expiration has passed. When unlocking multiple
+loans, invalid-before can be set to the latest claim expiration. If all Active UTxOs in the
+transaction are Finished Active UTxOs, the invalid-before bound is not necessary.
+
+Any lost collateral UTxOs claimed with this redeemer will still count as defaults for the borrower.
+
+Invalid Active UTxOs (ie, UTxOs with an `ActiveDatum` but no beacons) can be spent using just the
+`Unlock` spending redeemer. The loan spending script will directly check for the borrower's
+approval.
+
+## Benchmarks and Fee Estimations (YMMV)
+
+**No CIPs or hard-forks are needed. This protocol works on the Cardano blockchain, as is.**
```

Comment:
I'm really happy to hear this & it answered a question I had while reading about some CIPs I'd reviewed on the following subjects, and so was reassured this works with current Cardano architecture:
- input / output ordering in transactions (as mentioned is important for payments / interest / update outputs)
- transaction composability
- any other changes to Plutus and/or the Ledger that might be needed for a full, efficient implementation